invenio_pages/services/schemas.py

```diff
@@ -9,17 +10,47 @@
 
 from datetime import timezone
 
+from flask import current_app, has_app_context
+from invenio_config.default import ALLOWED_HTML_ATTRS, ALLOWED_HTML_TAGS
 from marshmallow import Schema, fields
 from marshmallow_utils.fields import SanitizedHTML, TZDateTime
 
 
+class DynamicSanitizedHTML(SanitizedHTML):
+    """
+    Extends `SanitizedHTML` to dynamically set allowed HTML tags and attributes based on Flask app context.
+
+    Preventing working outside of application context error.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the DynamicSanitizedHTML instance.
+
+        This constructor sets the defaults for HTML tags and attributes to None,
+        allowing them to be dynamically assigned later based on the application context.
+        """
+        super().__init__(tags=None, attrs=None, *args, **kwargs)
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        """Deserialize value with dynamic HTML tags and attributes based on Flask app context or defaults."""
+        if has_app_context():
```

Comment:
The `if has_app_context():` might not be needed, I see that in [SanitizedHTML](https://github.com/inveniosoftware/marshmallow-utils/blob/master/marshmallow_utils/fields/sanitizedhtml.py#L39) we did not add it. Is it needed for the tests?